### PR TITLE
Initial openshift configuration for the file-monitor service (plus 2 …

### PR DESCRIPTION
**Pull request by sturufous**
I got a bit overzealous with turning warnings into exceptions in my last commit, so I reverted the missing paper name code on line 574 of FileMonitorAction.cs to a warning.

### CODE REVIEW

File: openshift/kustomize/services/filemonitor/base/config-map.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    version: 1.0.0
+    component: filemonitor-service
+    managed-by: kustomize
+    created-by: jeremy.foster
+type: Opaque
+data:
+  KAKFA_CLIENT_ID: Filemonitor
+  MAX_FAIL_LIMIT: "5"
+  TOPICS: BCNG, MELTWATER
+  FILE_PATH: /data/papers
```
Comment (Fosol) [inline openshift/kustomize/services/filemonitor/base/config-map.yaml:22]: This should be `VOLUME_PATH: /data`.

Comment (Fosol) [reply openshift/kustomize/services/filemonitor/base/config-map.yaml:22]: I just noticed that the Image service also has this incorrect config.  Other services also appear to have this incorrect.

File: openshift/kustomize/services/filemonitor/base/deploy.yaml [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+              valueFrom:
+                configMapKeyRef:
+                  name: filemonitor-service
+                  key: MAX_FAIL_LIMIT
+            - name: Service__Topics
+              valueFrom:
+                configMapKeyRef:
+                  name: filemonitor-service
+                  key: TOPICS
+            - name: Service__NotificationTopic
```
Comment (Fosol) [inline openshift/kustomize/services/filemonitor/base/deploy.yaml:117]: I don't believe this service needs this.

File: openshift/kustomize/services/filemonitor/overlays/dev/kustomization.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        value: 20m
+      - op: replace
+        path: /spec/template/spec/containers/0/resources/requests/memory
+        value: 250Mi
+      - op: replace
+        path: /spec/template/spec/containers/0/resources/limits/cpu
+        value: 100m
+      - op: replace
+        path: /spec/template/spec/containers/0/resources/limits/memory
+        value: 500Mi
```
Comment (Fosol) [inline openshift/kustomize/services/filemonitor/overlays/dev/kustomization.yaml:43]: Lower to `120Mi`

File: openshift/kustomize/services/filemonitor/overlays/dev/kustomization.yaml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    patch: |-
+      - op: replace
+        path: /spec/replicas
+        value: 1
+      - op: replace
+        path: /spec/template/spec/containers/0/resources/requests/cpu
+        value: 20m
+      - op: replace
+        path: /spec/template/spec/containers/0/resources/requests/memory
+        value: 250Mi
```
Comment (Fosol) [inline openshift/kustomize/services/filemonitor/overlays/dev/kustomization.yaml:37]: Lower to `80Mi`

File: openshift/kustomize/services/filemonitor/overlays/prod/kustomization.yaml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    patch: |-
+      - op: replace
+        path: /spec/replicas
+        value: 1
+      - op: replace
+        path: /spec/template/spec/containers/0/resources/requests/cpu
+        value: 20m
+      - op: replace
+        path: /spec/template/spec/containers/0/resources/requests/memory
+        value: 250Mi
```
Comment (Fosol) [inline openshift/kustomize/services/filemonitor/overlays/prod/kustomization.yaml:37]: Lower to `80Mi`

File: openshift/kustomize/services/filemonitor/overlays/test/kustomization.yaml [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    patch: |-
+      - op: replace
+        path: /spec/replicas
+        value: 1
+      - op: replace
+        path: /spec/template/spec/containers/0/resources/requests/cpu
+        value: 20m
+      - op: replace
+        path: /spec/template/spec/containers/0/resources/requests/memory
+        value: 250Mi
```
Comment (Fosol) [inline openshift/kustomize/services/filemonitor/overlays/test/kustomization.yaml:37]: Lower to `80Mi`

File: openshift/kustomize/services/filemonitor/overlays/prod/kustomization.yaml [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+      - op: replace
+        path: /spec/template/spec/containers/0/resources/limits/cpu
+        value: 100m
+      - op: replace
+        path: /spec/template/spec/containers/0/resources/limits/memory
+        value: 500Mi
```
Comment (Fosol) [inline openshift/kustomize/services/filemonitor/overlays/prod/kustomization.yaml:43]: Lower to `120Mi`

File: openshift/kustomize/services/filemonitor/overlays/test/kustomization.yaml [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+      - op: replace
+        path: /spec/template/spec/containers/0/resources/limits/cpu
+        value: 100m
+      - op: replace
+        path: /spec/template/spec/containers/0/resources/limits/memory
+        value: 500Mi
```
Comment (Fosol) [inline openshift/kustomize/services/filemonitor/overlays/test/kustomization.yaml:43]: Lower to `120Mi`

**Review by Fosol — CHANGES_REQUESTED** [review]

File: openshift/kustomize/services/filemonitor/base/config-map.yaml [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+  namespace: default
+  annotations:
+    description: Filemonitor service configuration settings
+  labels:
+    name: filemonitor-service
+    part-of: tno
+    version: 1.0.0
+    component: filemonitor-service
+    managed-by: kustomize
+    created-by: jeremy.foster
```
[the remark this answers is not in the document]
Comment (sturufous) [reply openshift/kustomize/services/filemonitor/base/config-map.yaml:16]: Not sure what format to use to add more than one name. Comma delimited causes a kustomize error.

File: openshift/kustomize/services/filemonitor/base/config-map.yaml [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+type: Opaque
+data:
+  KAKFA_CLIENT_ID: Filemonitor
+  MAX_FAIL_LIMIT: "5"
+  TOPICS: BCNG, MELTWATER
```
[the remark this answers is not in the document]
Comment (sturufous) [reply openshift/kustomize/services/filemonitor/base/config-map.yaml:21]: I removed the topics.

File: openshift/kustomize/services/filemonitor/base/config-map.yaml (hunk 1 above)
Comment (sturufous) [reply openshift/kustomize/services/filemonitor/base/config-map.yaml:22]: I changed it to FILE_PATH: /data.

File: openshift/kustomize/services/filemonitor/base/deploy.yaml (hunk 2 above)
Comment (sturufous) [reply openshift/kustomize/services/filemonitor/base/deploy.yaml:117]: I removed the Service__NotificationTopic section.

File: openshift/kustomize/services/filemonitor/overlays/dev/kustomization.yaml (hunk 4 above)
Comment (sturufous) [reply openshift/kustomize/services/filemonitor/overlays/dev/kustomization.yaml:37]: Done.

File: openshift/kustomize/services/filemonitor/overlays/prod/kustomization.yaml (hunk 5 above)
Comment (sturufous) [reply openshift/kustomize/services/filemonitor/overlays/prod/kustomization.yaml:37]: Done.

File: openshift/kustomize/services/filemonitor/overlays/test/kustomization.yaml (hunk 6 above)
Comment (sturufous) [reply openshift/kustomize/services/filemonitor/overlays/test/kustomization.yaml:37]: Done.

File: openshift/kustomize/services/filemonitor/overlays/prod/kustomization.yaml (hunk 7 above)
Comment (sturufous) [reply openshift/kustomize/services/filemonitor/overlays/prod/kustomization.yaml:43]: Done.

File: openshift/kustomize/services/filemonitor/overlays/test/kustomization.yaml (hunk 8 above)
Comment (sturufous) [reply openshift/kustomize/services/filemonitor/overlays/test/kustomization.yaml:43]: Done.

File: openshift/kustomize/services/filemonitor/base/config-map.yaml (hunk 1 above)
Comment (Fosol) [reply openshift/kustomize/services/filemonitor/base/config-map.yaml]: It should be `VOLUME_PATH: /data`

File: openshift/kustomize/services/filemonitor/base/config-map.yaml [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+    version: 1.0.0
+    component: filemonitor-service
+    managed-by: kustomize
+    created-by: jeremy.foster
+type: Opaque
+data:
+  KAKFA_CLIENT_ID: Filemonitor
+  MAX_FAIL_LIMIT: "5"
+  TOPICS:
+  FILE_PATH: /data
```
Comment (Fosol) [inline openshift/kustomize/services/filemonitor/base/config-map.yaml:22]: `VOLUME_PATH: /data`

File: openshift/kustomize/services/filemonitor/base/config-map.yaml [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+    part-of: tno
+    version: 1.0.0
+    component: filemonitor-service
+    managed-by: kustomize
+    created-by: jeremy.foster
+type: Opaque
+data:
+  KAKFA_CLIENT_ID: Filemonitor
+  MAX_FAIL_LIMIT: "5"
+  TOPICS:
```
Comment (Fosol) [inline openshift/kustomize/services/filemonitor/base/config-map.yaml:21]: Remove

File: openshift/kustomize/services/filemonitor/base/config-map.yaml [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+  FILE_PATH: /data
+  NOTIFICATION_TOPIC: ""
```
Comment (Fosol) [inline openshift/kustomize/services/filemonitor/base/config-map.yaml:23]: Not required I don't believe at this time

**Review by Fosol — CHANGES_REQUESTED** [review]

File: openshift/kustomize/services/filemonitor/base/config-map.yaml (hunk 1 above)
Comment (sturufous) [reply openshift/kustomize/services/filemonitor/base/config-map.yaml]: Fixed it.

File: openshift/kustomize/services/filemonitor/base/config-map.yaml (hunk 12 above)
Comment (sturufous) [reply openshift/kustomize/services/filemonitor/base/config-map.yaml:21]: Remove just the TOPICS entry?

**Pull request closed without merging**